### PR TITLE
ManifestMapper feature

### PR DESCRIPTION
**Pull request by rattuscz**
As we're using different asset generator in webpack ( https://github.com/ztoben/assets-webpack-plugin ) we need option to format the manifest array before it's stored in cache.

So this is my attempt at adding new option to provide this mapper for ManifestLoader

### CODE REVIEW

**Comment by jiripudil** [comment]
Hello, thanks for your contribution! At first glance, it looks great. I'll try to give it a proper look during the weekend 🤞

File: src/DI/WebpackExtension.php [hunk 1]
```diff
@@ -81,6 +82,11 @@ public function loadConfiguration(): void
 			throw new ConfigurationException('You need to specify the dev server URL.');
 		}
 
+		if ($config['manifest']['mapper'] !== NULL) {
+			if (!class_exists($config['manifest']['mapper'])) {
+				throw new ConfigurationException("Non-existent classname '{$config['manifest']['mapper']}'provided for ManifestMapper.");
```
Comment (jiripudil) [inline src/DI/WebpackExtension.php:87]: I don't think this is necessary, the DI container will throw an exception anyway if it cannot find the class.

File: src/Manifest/ManifestLoader.php [hunk 2]
```diff
@@ -19,10 +19,16 @@ class ManifestLoader
 	 */
 	private $directoryProvider;
 
+	/**
+	 * @var ?ManifestMapper
```
Comment (jiripudil) [inline src/Manifest/ManifestLoader.php:23]: I think the code would be more streamlined if the mapper wasn't nullable. This package could by default use a `ManifestPluginMapper` which would just `return $manifest` in the `map()` method. Could you implement that please?

**Review by jiripudil — CHANGES_REQUESTED** [review]
As I've said earlier, it looks great, thank you! I'd just provide a simple default implementation of the mapper – I think the code will look even better without all the null-checks :)

File: src/DI/WebpackExtension.php (hunk 1 above)
Comment (rattuscz) [reply src/DI/WebpackExtension.php:87]: Oki, I don't have much experience with writing DI extensions so any advice welcome :)
will change.

File: src/Manifest/ManifestLoader.php (hunk 2 above)
Comment (rattuscz) [reply src/Manifest/ManifestLoader.php:23]: OK, will make default Identity mapper.

**Pull request merged**